src/main/kotlin/org/rust/ide/refactoring/move/RsMoveTopLevelItemsDialog.kt

```diff
@@ -0,0 +1,127 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.refactoring.move
+
+import com.intellij.openapi.fileChooser.FileChooserDescriptorFactory
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.guessProjectDir
+import com.intellij.openapi.ui.TextComponentAccessor
+import com.intellij.openapi.ui.TextFieldWithBrowseButton
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.refactoring.RefactoringBundle.message
+import com.intellij.refactoring.ui.RefactoringDialog
+import com.intellij.refactoring.util.CommonRefactoringUtil
+import com.intellij.ui.components.JBTextField
+import com.intellij.ui.layout.panel
+import com.intellij.util.IncorrectOperationException
+import org.rust.lang.RsFileType
+import org.rust.lang.core.psi.ext.RsItemElement
+import org.rust.lang.core.psi.ext.RsMod
+import org.rust.openapiext.toPsiFile
+import java.awt.Dimension
+import java.io.File
+import javax.swing.JComponent
+
+class RsMoveTopLevelItemsDialog(
+    project: Project,
+    private val itemsToMove: List<RsItemElement>,
+    private val containingMod: RsMod
+) : RefactoringDialog(project, false) {
+
+    private val sourceFilePath: String = containingMod.containingFile.virtualFile.path
+    private val sourceFileField: JBTextField = JBTextField(sourceFilePath).apply { isEnabled = false }
+    private val targetFileChooser: TextFieldWithBrowseButton = createTargetFileChooser(project)
+    private val memberPanel: RsMemberSelectionPanel = createMemberInfoPanel()
+
+    private var searchForReferences: Boolean = true
+
+    init {
+        super.init()
+        title = "Move Module Items"
+    }
+
+    private fun createTargetFileChooser(project: Project): TextFieldWithBrowseButton {
+        val chooser = TextFieldWithBrowseButton(null, disposable)
+        val fileChooserDescriptor = FileChooserDescriptorFactory.createSingleFileDescriptor(RsFileType)
+            .withRoots(project.guessProjectDir())
+        chooser.addBrowseFolderListener("Choose Destination File", null, project,
+            fileChooserDescriptor, TextComponentAccessor.TEXT_FIELD_WHOLE_TEXT)
+
+        chooser.text = sourceFilePath
+        chooser.textField.caretPosition = sourceFilePath.removeSuffix(".rs").length
+        chooser.textField.moveCaretPosition(sourceFilePath.lastIndexOf('/') + 1)
+
+        return chooser
+    }
+
+    override fun createCenterPanel(): JComponent {
+        val panel = panel {
+            row("From:") {
+                sourceFileField(growX).withLargeLeftGap()
```

Comment:
`withLargeLeftGap` is available only since 2020.1
You can wait a little bit - we are going to drop 193 support very soon.
Or you can try to use our conditional compilation ["magic"](https://github.com/intellij-rust/intellij-rust/blob/master/CONTRIBUTING.md#platform-versions) and create separate methods for each platform.
I prefer the first variant

Reply:
Yes, thank you, I will wait

Reply:
#5479